airflow/contrib/utils/sendgrid.py

```diff
@@ -62,15 +71,18 @@ def send_email(to, subject, html_content, files=None, dryrun=False, cc=None, bcc
     mail.add_content(Content('text/html', html_content))
 
     # Add email attachment.
-    for fname in files or []:
+    for fname in files:
         basename = os.path.basename(fname)
+
         attachment = Attachment()
+        attachment.type = mimetypes.guess_type(basename)[0]
+        attachment.filename = basename
+        attachment.disposition = "attachment"
+        attachment.content_id = '<{}>'.format(basename)
+
         with open(fname, "rb") as f:
-            attachment.content = base64.b64encode(f.read())
-            attachment.type = mimetypes.guess_type(basename)[0]
-            attachment.filename = basename
-            attachment.disposition = "attachment"
-            attachment.content_id = '<%s>' % basename
+            attachment.content = base64.b64encode(f.read()).decode('utf8')
```

Comment:
Are we sure that the sendgrid api expects a `str` and not a `bytes`?

Reply:
Yes.  The sendgrid code turns this data into a python dict and then `json.dumps` it.  In python3, `bytes` can not be JSON-serialized:

```python
Python 3.4.2 (default, Oct  8 2014, 10:45:20)
[GCC 4.9.1] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json.dumps(b'welkrjeklwj')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.4/json/__init__.py", line 230, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python3.4/json/encoder.py", line 192, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.4/json/encoder.py", line 250, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: b'welkrjeklwj' is not JSON serializable
```